### PR TITLE
Starting version of website with docs pages and rules

### LINKED ISSUE #3993: Documentation update request
Capto_Capture 2021-10-12_02-20-03_PM 2021-10-12 14_24_52

Love you guys! Thanks for everything you do! But can you please not bury the recommended installation and configuration three clicks deep?

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
## PR Checklist

- [x] Addresses an existing issue: fixes #3993 _(and will open many more followups)_
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/typescript-eslint/typescript-eslint/blob/master/CONTRIBUTING.md) were taken

## Overview

A trimmed down version of @armano2's beautiful #3147, with the `website/` directory moved into `packages/` for consistency. Builds off the same branch's Git commit history.

In addition to the tasks already mentioned in #3147, we'll need to follow up on at the very least:

- Deploying to Netlify
- Adding back the playground/REPL area
- Audit language, phrasing, flow, etc. of docs
- Reworking the rules page, such as improving the h1 and removing previous/next
- Accessibility/aXe audit
- Automate calling `generate-sponsors` regularly, such as the way `generate-contributors` is on a cron
- Fix the `warn Docs markdown link couldn't be resolved:` warnings
- Investigating moving `docs/` files to the `packages/website` directory _(not sure we want to)_

I'd additionally like to file followup tasks for:

- Search (Algolia?)
- Adding some kind of easter eggs :wink:
- Creating our own theme to stand out
- Requiring issues use the playground if possible 💪 

Unless directed otherwise I'll probably play with GitHub projects to set all that up.

### CODE REVIEW

File: website/webpack.plugin.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
+// const MonacoWebpackPlugin = require('monaco-editor-webpack-plugin');
```
Comment (JoshuaKGoldberg) [inline website/webpack.plugin.js:1]: I still have to go through and remove all these overrides & customizations. Will do once I get Node set up in the Windows area of my new laptop...

**Comment by armano2** [comment]
@JoshuaKGoldberg as for new compile errors i addressed it in #4101

File: website/webpack.plugin.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+const webpack = require('webpack');
+const path = require('path');
+
+module.exports = function (/*context, options*/) {
+  return {
+    name: 'webpack-custom-plugin',
+    configureWebpack(cfg, isServer) {
```
Comment (armano2) [inline website/webpack.plugin.js:8]: if we are not exposing playground here, we can/should remove this file

File: website/webpack.plugin.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+          ),
+          new webpack.NormalModuleReplacementPlugin(
+            /semver/,
+            path.resolve(__dirname, 'src/modules/semver.js'),
+          ),
+          new webpack.NormalModuleReplacementPlugin(
+            /is-glob/,
+            path.resolve(__dirname, 'src/modules/is-glob.js'),
+          ),
+        ],
```
Comment (armano2) [inline website/webpack.plugin.js:114]: if we are not exposing playground here, we can/should remove this file

or atleast remove all content

```suggestion
```

File: website/src/modules/eslint-rules.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+import builtinRules from 'eslint/lib/rules';
```
Comment (armano2) [inline website/src/modules/eslint-rules.js:1]: this file can be removed (with all files in this folder)

File: website/package.json [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    "util": "0.12.4"
+  },
+  "devDependencies": {
+    "@docusaurus/module-type-aliases": "^2.0.0-beta.9",
+    "@types/react-helmet": "^6.1.4",
+    "@types/react-router-dom": "^5.3.2",
+    "@types/react": "^17.0.34",
+    "globby": "^11.0.4",
+    "monaco-editor": "^0.29.1",
+    "string-replace-loader": "^3.0.3",
```
Comment (armano2) [inline website/package.json:45]: to be removed

File: website/docusaurus.config.js [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+const lightCodeTheme = require('prism-react-renderer/themes/github');
+const darkCodeTheme = require('prism-react-renderer/themes/dracula');
+
+const sponsors = require('./data/sponsors.json');
+
+const remarkPlugins = [
+  [require('@docusaurus/remark-plugin-npm2yarn'), { sync: true }],
+];
+
+const githubUrl = 'https://github.com/armano2/typescript-eslint';
```
Comment (armano2) [inline website/docusaurus.config.js:13]:
```suggestion
const githubUrl = 'https://github.com/typescript-eslint/typescript-eslint';
```

File: packages/eslint-plugin/tests/docs.test.ts [hunk 7]
```diff
@@ -57,14 +57,23 @@ describe('Validating rule docs', () => {
...
         gfm: true,
         silent: false,
       });
 
+      expect(tokens[0]).toMatchObject({ type: 'hr', raw: '---\n' });
+      expect(tokens[1]).toMatchObject({
+        text: `hide_title: true\nsidebar_label: ${ruleName}`,
+        type: 'paragraph',
+      });
+      expect(tokens[2]).toMatchObject({ type: 'hr', raw: '---\n\n' });
```
Comment (armano2) [inline packages/eslint-plugin/tests/docs.test.ts:72]: to be removed

File: website/src/vendor/ds/createDesignSystem.d.ts [hunk 8]
```diff
@@ -0,0 +1,71 @@
+import type { Sandbox } from '../sandbox';
```
Comment (armano2) [inline website/src/vendor/ds/createDesignSystem.d.ts:1]: vendor directory should be rmeoved

File: website/package.json [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+    "@typescript-eslint/visitor-keys": "5.3.0",
+    "assert": "2.0.0",
+    "clsx": "^1.1.1",
+    "eslint": "*",
+    "os-browserify": "0.3.0",
+    "path-browserify": "1.0.1",
+    "react-dom": "^17.0.2",
+    "react": "^17.0.2",
+    "typescript": "*",
+    "util": "0.12.4"
```
Comment (armano2) [inline website/package.json:36]: to be removed

File: website/package.json [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+    "@typescript-eslint/parser": "5.3.0",
+    "@typescript-eslint/scope-manager": "5.3.0",
+    "@typescript-eslint/types": "5.3.0",
+    "@typescript-eslint/typescript-estree": "5.3.0",
+    "@typescript-eslint/visitor-keys": "5.3.0",
+    "assert": "2.0.0",
+    "clsx": "^1.1.1",
+    "eslint": "*",
+    "os-browserify": "0.3.0",
+    "path-browserify": "1.0.1",
```
Comment (armano2) [inline website/package.json:32]: to be removed

File: website/package.json [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+    "@docusaurus/remark-plugin-npm2yarn": "^2.0.0-beta.9",
+    "@docusaurus/theme-classic": "^2.0.0-beta.9",
+    "@mdx-js/react": "1.6.22",
+    "@typescript-eslint/eslint-plugin": "5.3.0",
+    "@typescript-eslint/parser": "5.3.0",
+    "@typescript-eslint/scope-manager": "5.3.0",
+    "@typescript-eslint/types": "5.3.0",
+    "@typescript-eslint/typescript-estree": "5.3.0",
+    "@typescript-eslint/visitor-keys": "5.3.0",
+    "assert": "2.0.0",
```
Comment (armano2) [inline website/package.json:28]: to be removed

File: website/package.json (hunk 5 above)
Comment (armano2) [inline website/package.json:45]: to be removed
```suggestion
```

**Review by armano2 — CHANGES_REQUESTED** [review]
`website/src/modules` and `website/src/vendor` should be removed

File: website/package.json [hunk 12]
```diff
@@ -25,24 +25,18 @@
...
     "@typescript-eslint/typescript-estree": "5.3.0",
     "@typescript-eslint/visitor-keys": "5.3.0",
-    "assert": "2.0.0",
     "clsx": "^1.1.1",
     "eslint": "*",
-    "os-browserify": "0.3.0",
-    "path-browserify": "1.0.1",
     "react-dom": "^17.0.2",
     "react": "^17.0.2",
     "typescript": "*",
```
Comment (armano2) [inline website/package.json:32]:
```suggestion
    "typescript": "*"
```

**Comment by JoshuaKGoldberg** [comment]
https://github.com/typescript-eslint/typescript-eslint/runs/4133112493?check_suite_focus=true

```
> nx run website:build 
$ docusaurus build

Minimum Node version not met :(

You are using Node v12.22.7. We require Node >=14 or up!
```

Grr.

**Pull request closed without merging**